lib/api/client-commands/_base-command.ts

```diff
@@ -1,5 +1,22 @@
-module.exports = class ClientCommand {
-  static get isTraceable() {
+interface ResultObject {
+  status: number;
+  code: any;
+  name: any;
+  value: {
+    message: any;
+  };
+  error: any;
+}
+
+interface MakePromiseOptions {
+  performAction: (callback : (result : ResultObject) => void) => void;
+  userSuppliedCallback?: (param : any) => any;
+  fullResultObject?: boolean;
+  fullPromiseResolve?: boolean;
+}
+
+export default class ClientCommand {
```

Comment:
Should be `export = class ClientCommand`, if it is being imported in other JS files, so that we don't need to make any changes in those JS files. (`export =` is similar to `module.exports =` in JS)

^ is the reason why the tests are failing.

And when converting those JS files (in which ClientCommand is being used) to TS, we will use `import ClientCommand = require('./_base-command.js');` instead of `import ClientCommand from './_base-command.js';`.

Once we are sure that we are importing this file in TS files only, we can change the export back to `export default class ClientCommand`, and the imports in other TS files to `import ClientCommand from './_base-command.js';`.